### PR TITLE
Add Envoy support for HTTP/1.0 health-check requests

### PR DESCRIPTION
**Pull request by ctrlaltluc**
| Q               | A      |
| --------------- | ------ |
| New feature?    | yes |

### What's in this PR?

Added Envoy support for HTTP/1.0 health-check requests.

### Why?

Some older Load Balancers have default health-checks in HTTP/1.0, and when those are using the Envoy health-check endpoints without HTTP/1.0 support, they fail. 

### Checklist

- [x] Implementation tested
- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/developer-guide/blob/master/docs/coding-style/error-handling-guide.md)
- [x] Logging code meets the guideline
- [x] User guide and development docs updated (if needed)

### CODE REVIEW

**Review by stoader — COMMENTED** [review]
According to the documentation https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/protocol.proto 

```
This is off by default, and not fully standards compliant. There is support for pre-HTTP/1.1 style connect logic, dechunking, and handling lack of client host iff default_host_for_http_10 is configured.
```

So I'd make the use of this config optional through the KafkaCluster CR instead of enabling it by default. Users would have to enable it explicitly.

@amuraru what do you think?

**Comment by amuraru** [comment]
Fair point - we can put this behind a feature flag, right @ctrlaltluc ?

**Comment by ctrlaltluc** [comment]
Thanks for your comments!

Sure @amuraru. Just to double check, I think that @stoader was referring to leave it false by default and let clients pass it via operator config. Am I mistaken @stoader?

**Comment by ctrlaltluc** [comment]
Thinking about it, it is basically the same as you meant @amuraru. I will amend the PR.

**Comment by amuraru** [comment]
IIUC the proposal is to have the KafkaCluster CRD extended to support an optional flag for this.
Something like `EnableHttp10ForHealthCheckPort` (probably shorter :) ) in: https://github.com/banzaicloud/koperator/blob/1975aebe9020f2009a96ffc416fbe7db7b2a1623/api/v1beta1/kafkacluster_types.go#L268-L297

**Comment by stoader** [comment]
> IIUC the proposal is to have the KafkaCluster CRD extended to support an optional flag for this. Something like `EnableHttp10ForHealthCheckPort` (probably shorter :) ) in:
> 
> https://github.com/banzaicloud/koperator/blob/1975aebe9020f2009a96ffc416fbe7db7b2a1623/api/v1beta1/kafkacluster_types.go#L268-L297

I wonder if this flag can be enabled or not through the [CommandLineArgs](https://github.com/banzaicloud/koperator/blob/1975aebe9020f2009a96ffc416fbe7db7b2a1623/api/v1beta1/kafkacluster_types.go#L296)

**Comment by amuraru** [comment]
> I wonder if this flag can be enabled or not through the [CommandLineArgs](https://github.com/banzaicloud/koperator/blob/1975aebe9020f2009a96ffc416fbe7db7b2a1623/api/v1beta1/kafkacluster_types.go#L296)

Just checked that but envoy doesn't support enabling this via command line arguments

**Comment by panyuenlau** [comment]
@ctrlaltluc Just curious if there are any updates on this PR? Or should we take it from here and update the PR with the suggestions from @stoader and @amuraru?

**Comment by ctrlaltluc** [comment]
@panyuenlau I will follow up next week with a new commit. I am on holiday this week. I had a commit prepared, just needed to change some tests too.

**Comment by panyuenlau** [comment]
> 

Thanks for the update and look forward to the updated PR!

**Comment by panyuenlau** [comment]
> @panyuenlau I will follow up next week with a new commit. I am on holiday this week. I had a commit prepared, just needed to change some tests too.

@ctrlaltluc Thanks for the update and we look forward to the update!

File: api/v1beta1/kafkacluster_types.go [hunk 1]
```diff
@@ -321,6 +321,8 @@ type EnvoyConfig struct {
 	// If not specified, the Envoy pods' priority is default to zero
 	// +optional
 	PriorityClassName string `json:"priorityClassName,omitempty"`
+	// EnableHealthCheckHttp10 is a toggle for adding HTTP1.0 support to Envoy health-check
```
Comment (panyuenlau) [inline api/v1beta1/kafkacluster_types.go:324]: I would add `+optional` just to make it more explicit to the readers that this newly added field is optional
```suggestion
	// EnableHealthCheckHttp10 is a toggle for adding HTTP1.0 support to Envoy health-check
	// +optional
```

**Review by panyuenlau — CHANGES_REQUESTED** [review]
Thank you for the contribution!

Can you please update this [`envoyConfig` ](https://github.com/banzaicloud/koperator/blob/master/config/samples/banzaicloud_v1beta1_kafkacluster.yaml#L238) section about this newly added field? This sample yaml serves as a living documentation about most (if not all) of the fields that we have for `KafkaCluster` custom resource

Also, please sign the CLA(Contributor License Agreement) to complete the check that we placed for first-time contributors

**Comment by panyuenlau** [comment]
Can you please update [this sample KafkaCluster yaml](https://github.com/banzaicloud/koperator/blob/master/config/samples/banzaicloud_v1beta1_kafkacluster.yaml#L238) to reflect the newly added field?

**Comment by ctrlaltluc** [comment]
> @ctrlaltluc Can you please update [this sample KafkaCluster yaml](https://github.com/banzaicloud/koperator/blob/master/config/samples/banzaicloud_v1beta1_kafkacluster.yaml#L238) to reflect the newly added field?

Managed to add it in [48a78f3](https://github.com/banzaicloud/koperator/pull/830/commits/48a78f36bff35c9126e7b14208e3bb19c94f1252).

**Review by stoader — APPROVED** [review]

**Review by pregnor — APPROVED** [review]

**Review by panyuenlau — APPROVED** [review]

**Comment by ctrlaltluc** [comment]
@panyuenlau the PR is ready to be merged now. :)

**Comment by pregnor** [comment]
I just wanted to ask that, thanks for confirming 🙂

**Pull request merged**